game/3dobject.cc

```diff
@@ -99,7 +99,7 @@ void Object3d::loadWavefrontObj(fs::path const& filepath, float scale) {
 }
 
 void Object3d::load(fs::path const& filepath, fs::path const& texturepath, float scale) {
-	if (!texturepath.empty()) m_texture.reset(new Texture(texturepath));
```

Comment:
sptr.reset(new Foo) does create new Foo, delete old *sptr and store the new one in sptr. This is essentially the same as with sptr = make_unique: create new (into a temporary unique_ptr), then move-assign that into sptr, deleting any existing *sptr. The biggest difference is when something throws because a plain pointer may be leaked but make_unique keeps it safer. Also, the syntax is nicer with make_unique.